vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/KnowledgeBase.java

```diff
@@ -220,6 +256,74 @@ public void addStatements(Statement... statements) {
 		notifyListenersOnStatementsAdded(addedStatements);
 	}
 
+	/**
+	 * Removes a single statement from the knowledge base.
+	 * 
+	 * @return true, if the knowledge base has changed.
+	 * @param statement
+	 */
+	public void removeStatement(Statement statement) {
+		if (doRemoveStatement(statement)) {
+			notifyListenersOnStatementRemoved(statement);
+		}
+	}
+
+	/**
+	 * Adds a single statement to the knowledge base.
+	 * 
+	 * @param statement the statement to be added
+	 * @return true, if the knowledge base has changed.
+	 */
+	boolean doRemoveStatement(Statement statement) {
+		Validate.notNull(statement, "Statement cannot be Null!");
+
+		if (this.statements.contains(statement) && statement.accept(this.removeStatementVisitor)) {
+			this.statements.remove(statement);
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Removes a collection of statements to the knowledge base.
+	 * 
+	 * @param statements
```

Comment:
Should we describe the parameter, instead of just name it?

Reply:
done